### PR TITLE
sponsorship pm pnd len reduction

### PR DESCRIPTION
**Pull request by livingrockrises**
# Summary

L1 calldata reduction by reducing pnd length more than 50%

Related Issue: # (issue number)

## Change Type

- [ ] Bug Fix
- [ ] Refactor
- [ ] New Feature
- [ ] Breaking Change
- [ ] Documentation Update
- [ ] Performance Improvement
- [ ] Other

# Checklist

- [ ] My code follows this project's style guidelines
- [ ] I've reviewed my own code
- [ ] I've added comments for any hard-to-understand areas
- [ ] I've updated the documentation if necessary
- [ ] My changes generate no new warnings
- [ ] I've added tests that prove my fix is effective or my feature works
- [ ] All unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published

# Additional Information

# Branch Naming

### CODE REVIEW

File: contracts/sponsorship/SponsorshipPaymaster.sol [hunk 1]
```diff
@@ -233,11 +233,10 @@ contract SponsorshipPaymaster is
...
     {
-        (paymasterId, validUntil, validAfter, priceMarkup) = abi.decode(
-            paymasterAndData[VALID_PND_OFFSET:SIGNATURE_OFFSET],
-            (address, uint48, uint48, uint32)
-        );
-
+        paymasterId = address(bytes20(paymasterAndData[VALID_PND_OFFSET:VALID_PND_OFFSET+20]));
+        validUntil = uint48(bytes6(paymasterAndData[40:46]));
+        validAfter = uint48(bytes6(paymasterAndData[46:52]));
+        priceMarkup = uint32(bytes4(paymasterAndData[52:56]));
```
Comment (ankurdubey521) [inline contracts/sponsorship/SponsorshipPaymaster.sol:239]: should these be in terms of `VALID_PND_OFFSET` as mentioned above?

Comment (livingrockrises) [reply contracts/sponsorship/SponsorshipPaymaster.sol:239]: yeah I was thinking what would be the best?
VALID_PND_OFFSET, VALID_PND_OFFSET+20
VALID_PND_OFFSET+20, VALID_PND_OFFSET+20+6

and so on

then I don't really need SIGNATURE_OFFSET also

or I can remove offsets all together and stick to this numbers only 
wdyt? @ankurdubey521

Comment (ankurdubey521) [reply contracts/sponsorship/SponsorshipPaymaster.sol:239]: I think the former is fine

Comment (livingrockrises) [reply contracts/sponsorship/SponsorshipPaymaster.sol]: done

**Review by AmanRaj1608 — APPROVED** [review]

**Review by tomarsachin2271 — APPROVED** [review]

**Pull request merged**